client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java

```diff
@@ -273,6 +274,69 @@ public static boolean setConnectionInfo(int inst, BookmarkBase bookmark) {
         String[] arrayArgs = args.toArray(new String[args.size()]);
         return freerdp_parse_arguments(inst, arrayArgs);
     }
+    
+    public static boolean setConnectionInfo(int inst, Uri openUri) {
+        ArrayList<String> args = new ArrayList<String>();
+
+        // Parse URI from query string. Same key overwrite previous one
+        // freerdp://user@ip:port/connect?sound=&rfx=&p=password&clipboard=%2b&themes=-
+     
+        // Now we only support Software GDI
+        args.add(TAG);
+        args.add("/gdi:sw");
+        
+        // Parse hostname and port. Set to 'v' argument
+        String hostname = openUri.getHost();
+        int port = openUri.getPort();
+        if (hostname != null) {
+            hostname = hostname + ((port == -1) ? "" : (":" + String.valueOf(port)));
+            args.add("/v:" + hostname);
+        }
+        
+        String user = openUri.getUserInfo();
+        if (user != null) {
+            args.add("/u:" + user);
+        }
+        
+        for (String key: openUri.getQueryParameterNames()) {
+            String value = openUri.getQueryParameter(key);
+            
+            if (value.isEmpty()) {
+                // Query: key=
+                // To freerdp argument: /key
+                args.add("/" + key);
+            } else if (value.equals("-") || value.equals("+")) {
+                // Query: key=- or key=+
+                // To freerdp argument: -key or +key
+                args.add(value+key);
+            } else {
+                // Query: key=value
+                // To freerdp argument: /key:value
+                if (key.equals("drive") && value.equals("sdcard")) { 
+                    // Special for sdcard redirect
+                    String path = android.os.Environment.getExternalStorageDirectory().getPath();
+                    args.add("/drive:sdcard," + path);
+                } else {
+                    args.add("/" + key + ":" + value);
+                }
+            }
+        }
+        
+        // Build string for argument tracing. Hide password
+        StringBuilder sbArguments = new StringBuilder();
+        for (String argument: args) {
+            if (argument.startsWith("/p:")) {
+                sbArguments.append("/p:*");
+            } else {
+                sbArguments.append(argument);
+            }
+            sbArguments.append(" ");
+        }
+        Log.v(TAG, "Connecting with arguments: " + sbArguments.toString());
```

Comment:
Yeah you are exactly correct. I just checked command line and find following password arguments.
    /pth:<password hash>        Pass the hash (restricted admin mode)
    /p:<password>               Password
    /gp:<password>              Gateway password
    /assistance:<password>      Remote assistance password

I haven't notice the new command line argument these days :). I agree to totally discard android logging for arguments. Because we don't know whether there will be more arguments future.